src/path.rs

```diff
@@ -10,7 +10,7 @@ use thiserror::Error;
 #[derive(Clone, Debug, PartialEq)]
```

Comment:
```suggestion
// TODO: it might be useful to split this into CidPath and IpnsPath, then have Ipns resolve through
// latter into CidPath (recursively) and have dag.rs support only CidPath. Keep IpfsPath as a
// common abstraction which can be either.
#[derive(Clone, Debug, PartialEq)]
```